src/Pages/CountryDetail.jsx

```diff
@@ -1,151 +1,172 @@
+                to={`/Details/${border.name}`}
+              >
+                {border.name}
+              </Link>
+            ))}
+          </div>
+        </div>
+      </main>
+
+      <footer className="text-center text-balance p-4">
+        <p className="text-sm">
+          Challenge by{" "}
+          <a className="text-blue-700" href="frontendmentor.io">
+            Frontend Mentor.
+          </a>{" "}
+          coded by{" "}
+          <a className="text-blue-700" href="https://github.com/Mazz100">
+            Mazen Hassan.
+          </a>
+        </p>
+      </footer>
+    </div>
+  );
 }
 
-
 //Exporting an asyc function loader and use it inside react router path
 
 export const countryDetailsLoader = async ({ params }) => {
-    const { countryId } = params;
-
-    try {
-
-        const countryDetail = await fetch(`https://restcountries.com/v3.1/name/${countryId}?fullText=true&fields=flags,name,nativeName,population,region,subregion,capital,tld,currencies,languages,borders`)
-            .then(res => res.json())
-            .then(countriesData => countriesData.map(country => ({
-                flags: country.flags.svg,
-                flagAlt: country.flags.alt,
-                name: country.name.common,
-                nativeName: Object.entries(country.name.nativeName)
-                    //Mapping through nativeName values with unique key
-                    .map(([key, value]) => ` ${value.official}`)
-                    .join(', '),
-                population: country.population.toLocaleString(),
-                region: country.region,
-                subRegion: country.subregion,
-                topLevelDomain: country.tld,
-                capital: country.capital,
-                languages: Object.values(country.languages).join(', '),
-                capital: country.capital,
-                currency: Object.entries(country.currencies).map(([key, value]) =>
-                    `${value.name}`).join(', '),
-                borders: country.borders,
-            })))
-
-
-        const borderNamesQuery = countryDetail.map(country => country.borders).join(',') //alpha codes expect comma separator
-
-
-        const borderNames = borderNamesQuery && await fetch(`https://restcountries.com/v3.1/alpha?codes=${borderNamesQuery}?fields=name,cca3`)
-            .then(res => res.json())
-            .then(countries => countries.map(country => ({ name: country.name.common, code: country.cca3 })))
-
-
-        return [{ ...countryDetail, borderNames }]
-    }
-
-
-    catch (error) {
-        console.log(error);
-    }
-
-
-}
-
+  const { countryId } = params;
+
+  try {
+    const [rawCountry] = await fetch(
+      `https://restcountries.com/v3.1/name/${countryId}?fullText=true&fields=flags,name,nativeName,population,region,subregion,capital,tld,currencies,languages,borders`
+    ).then((res) => res.json());
+
+    const bordersQuery = rawCountry.borders.join(",");
+
+    const borders = await fetch(
```

Comment:
Fetching borders is simple. It returns an array, but that's what we want in this case, a list of all the bordering countries